analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,186 @@
+compareClinVarToPathway <- function(pathwayName=NULL, clinVarName=NULL)
+{
+	tmpDat <- data.frame(clinData[,clinVarName], geneSetExpMat[pathwayName,])
+	colnames(tmpDat) <- c("ClinVar", "PathwayVar") 	#Need to set this since pathwayName and clinVarName will change which is not good for aov function
+	aovOut <- aov(PathwayVar ~ ClinVar, data=tmpDat)
+	posthoc <- TukeyHSD(x=aovOut, conf.level=0.95)
+	posthoc <- data.frame(posthoc[[1]])
+	aovOutNamed <- unlist(summary(aovOut))
+	posthoc[,"F_Value"] <- aovOutNamed["F value1"]
+	posthoc[,"AOV_P"] <- aovOutNamed["Pr(>F)1"]
+	posthoc[,"Pathway"] <- pathwayName
+	posthoc[,"ClinVar"] <- clinVarName
+	posthoc <- cbind(t(sapply(rownames(posthoc), FUN=getVarNames)), posthoc) #Pulls out disease names 
+	colnames(posthoc)[1:2] <- c("VarX", "VarY")
+	return(posthoc)
+}
+
+
+#Analysis by short_histology. This is only clinical variable we are examining but the function compareClinVarToPathway
+#is generic enough that any variable in clinical data can be used. This will be useful if we decide to use for instance molecular
+#subtypes instead of histology for this analysis. So the lapply below will for each pathway
+#run an ANOVA across all histologies i.e. looking to see if any particular histology has a significantly greater score
+#than any other histology and then follow that with a post-hoc test. The comparison is not done between pathways. 
+diseaseType <- lapply(rownames(geneSetExpMat), FUN=compareClinVarToPathway, clinVarName="short_histology")
+diseaseType <- do.call("rbind", diseaseType) 
+
+#Filter to significant entries by F and Tukey and write out 
+diseaseTypeFilt <- diseaseType %>% dplyr::filter( AOV_P <= aovParam, p.adj <= padjParam)
+colnames(diseaseTypeFilt)[1:2] <- c("DiseaseX", "DiseaseY") #Rename columns to match clinical variable type
+write.table(diseaseTypeFilt, "results/DiseaseCorrelationPathway.txt", sep="\t", row.names=F)
+
+#We would like to filter to cases where a histology is significantly higher than other histologies (and the corresponding pathways)
+#This is for the heatmap so we may focus a bit on the more interesting results
+#The issue here is that Tukey will write out the pathways in some particular order
+#If we are interested in finding if a Histology A is significantly upregulated compared to at least X% of
+#other histologies this may present a problem because Tukey may write out the following
+#HistologyA-HistologyB ~ Fold-change = 1
+#HistologyA-HistologyC ~ Fold-change = 1
+#HistologyD-HistologyA ~ Fold-change = -1
+#if the first histology is in column 1 and the second is in column 2 and the FC is in column 3 of a data frame we would be required
+#to check the FC in order to determine if Histology Y is in fact > or < another histology. Hence a bit of logic below to get counts. 
+diseaseTypeFilt[,"DiseaseTmpX"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"DiseaseX"]), as.character(diseaseTypeFilt[,"DiseaseY"]))
```

Comment:
The column `DiseaseTmpX` will contain the histology with higher expression, did I follow correctly? If so, can you call this something more descriptive than `DiseaseTmpX` please?

Reply:
Sure, updating, see comment below.